internals/daemon/api_exec_test.go

```diff
@@ -108,6 +108,28 @@ func (s *execSuite) TestEnvironment(c *C) {
 	c.Check(stderr, Equals, "")
 }
 
+func (s *execSuite) TestEnvironmentInheritedFromDaemon(c *C) {
+	oldEnv, envWasSet := os.LookupEnv("FOO")
+	err := os.Setenv("FOO", "bar")
+	c.Check(err, IsNil)
+	defer func() {
+		var err error
+		if envWasSet {
+			err = os.Setenv("FOO", oldEnv)
+		} else {
+			err = os.Unsetenv("FOO")
+		}
+		c.Check(err, IsNil)
+	}()
+
+	stdout, stderr, waitErr := s.exec(c, "", &client.ExecOptions{
+		Command: []string{"/bin/sh", "-c", "echo FOO=$FOO"},
```

Comment:
you should also test when `exec` has its own Environment

Reply:
Added in https://github.com/canonical/pebble/pull/234/commits/24bb4d2db1ac721b907b4c30c4bd1bd71ff199f4. Thanks!